### PR TITLE
ENH: Update docstring of class KrylovJacobian to fix #2744

### LINKED ISSUE #2744: BUG: optimize: '**kw' argument of 'newton_krylov' is not documented
The signature for `optimize.newton_krylov` includes `**kw`, but `kw` is not documented in the `Parameters` section of the docstring.  The docstring should explain what is allowed, or at least explain how to find out (e.g. "Additional keyword arguments are passed to `foo`; see the `foo` docstring for more details.").

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix #2744

#### What does this implement/fix?
update docstring of class KrylovJacobian.

#### Additional information
This PR #2787 is no longer active, so I created this PR based on the discussion of #2787.

### CODE REVIEW

File: scipy/optimize/nonlin.py [hunk 1]
```diff
@@ -1386,12 +1390,13 @@ class KrylovJacobian(Jacobian):
...
         as ``update(x, f)`` after each nonlinear step, with ``x`` giving
         the current point, and ``f`` the current function value.
-    inner_tol, inner_maxiter, ...
-        Parameters to pass on to the \"inner\" Krylov solver.
-        See `scipy.sparse.linalg.gmres` for details.
     outer_k : int, optional
         Size of the subspace kept across LGMRES nonlinear iterations.
         See `scipy.sparse.linalg.lgmres` for details.
+    inner_kwargs : kwargs
+        Keyword parameters for the \"inner\" Krylov solver (defined with `method`). Parameter names must start
```
Comment (ilayn) [inline scipy/optimize/nonlin.py]: These lines are longer than 80 characters

Comment (AtsushiSakai) [reply scipy/optimize/nonlin.py]: Thanks I fixed it.

File: scipy/optimize/nonlin.py [hunk 2]
```diff
@@ -1373,6 +1373,10 @@ class KrylovJacobian(Jacobian):
         the iterative solvers in `scipy.sparse.linalg`.
 
         The default is `scipy.sparse.linalg.lgmres`.
+    inner_maxiter : int, optional
+        Parameter to pass to the \"inner\" Krylov solver: maximum number of
```
Comment (ilayn) [inline scipy/optimize/nonlin.py]: Inside triple quote strings `"""..."""` you don't need to escape double quotes.

File: scipy/optimize/nonlin.py [hunk 3]
```diff
@@ -1386,12 +1390,14 @@ class KrylovJacobian(Jacobian):
...
         as ``update(x, f)`` after each nonlinear step, with ``x`` giving
         the current point, and ``f`` the current function value.
-    inner_tol, inner_maxiter, ...
-        Parameters to pass on to the \"inner\" Krylov solver.
-        See `scipy.sparse.linalg.gmres` for details.
     outer_k : int, optional
         Size of the subspace kept across LGMRES nonlinear iterations.
         See `scipy.sparse.linalg.lgmres` for details.
+    inner_kwargs : kwargs
+        Keyword parameters for the \"inner\" Krylov solver
```
Comment (ilayn) [inline scipy/optimize/nonlin.py]: Same quote comment here.

**Review by ilayn — COMMENTED** [review]
Minor comments

File: scipy/optimize/nonlin.py (hunk 2 above)
Comment (AtsushiSakai) [reply scipy/optimize/nonlin.py]: I fixed it.

File: scipy/optimize/nonlin.py (hunk 3 above)
Comment (AtsushiSakai) [reply scipy/optimize/nonlin.py]: I fixed it.

**Comment by ilayn** [comment]
Looks good. Do you mind adding an example for these while we are at it?

**Review by ilayn — APPROVED** [review]

**Comment by AtsushiSakai** [comment]
Do you mean an example of using KrylovJacobian?. I'm sorry I am not familiar with it....

**Comment by ilayn** [comment]
Then nevermind. Thanks @AtsushiSakai

**Pull request merged**